### ISSUE TITLE
Overload `AddHangfireServer` to allow configuration of BackgroundJobServerOptions

### ISSUE THREAD

**Issue by coolhome** [HangfireIO/Hangfire#1463]
It would be really convenient to add an overload to `HangfireServiceCollectionExtensions.AddHangfireServer` to allow configuration of `BackgroundJobServerOptions` like `AddHangfire` and `GlobalConfiguration`

https://github.com/HangfireIO/Hangfire/blob/master/src/Hangfire.AspNetCore/HangfireServiceCollectionExtensions.cs#L125

Current Solution:
```cs
services.AddSingleton(provider => new BackgroundJobServerOptions
{
    Queues = new[] { "default", "misc" },
    WorkerCount = 1,
});
services.AddHangfireServer();
```

Proposed Solution:
```cs
services.AddHangfireServer(options =>
{
    options.WorkCount = 1;
    options.Queues = new[] {"default", "misc"};
});
```

**Comment by stevendesrochers** [comment]
I had the [same issue](https://stackoverflow.com/questions/56112793/is-there-a-way-to-setup-queue-listening-for-hangfire-in-the-configureservices/56112933) and wondered why there was no overload too. Would be a nice addition!

**Comment by djesmond** [comment]
According to the [docs](https://docs.hangfire.io/en/latest/background-processing/configuring-queues.html) it is an option. But it is done in the `Configure` method and not `ConfigureServices` in the `Startup` file of core. However using this approach is not desribed in [Getting Started](https://docs.hangfire.io/en/latest/getting-started/aspnet-core-applications.html). I do not know what difference, if any, it makes but it might be worth a try.

**Comment by coolhome** [comment]
@djesmond In Hangfire 1.7 the `AddHangfireServer` was introduced to add a [`IHostedService`](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/generic-host?view=aspnetcore-2.2) to WebHost and GenericHost of ASP.NET Core 2.1+.

I believe `UseHangfireServer` was built around using WebHost `IApplicationLifeTime` for starting and stopping. Not as clean imho

**Comment by djesmond** [comment]
I see. So `AddHangfireServer` can be considered the replacement for `UseHangfireServer`?

**Comment by odinserj** [comment]
Indeed much simpler, thanks. Just added an overload that will be available in upcoming 1.7.5.
